packages/server/src/asyncQuestion/asyncQuestion.controller.ts

```diff
@@ -150,8 +158,98 @@ export class asyncQuestionController {
     }
   }
 
-  @Patch(':questionId')
-  async updateQuestion(
+  @Patch('student/:questionId')
+  /**
+   * Updates a student's async question.
+   *
+   * @param {number} questionId - The ID of the question to update.
+   * @param {UpdateAsyncQuestions} body - The updated question data.
+   * @param {UserModel} user - The user making the request.
+   * @return {Promise<AsyncQuestionParams>} The updated question.
+   * @throws {NotFoundException} If the question is not found.
+   * @throws {HttpException} If the user is not the creator of the question.
+   * @throws {HttpException} If the user tries to update the question's status to TADeleted or HumanAnswered.
+   */
+  async updateStudentQuestion(
+    @Param('questionId') questionId: number,
+    @Body() body: UpdateAsyncQuestions,
+    @User() user: UserModel,
+  ): Promise<AsyncQuestionParams> {
+    const question = await AsyncQuestionModel.findOne({
+      where: { id: questionId },
+      relations: ['course', 'creator', 'votes'],
+    });
+
+    if (!question) {
+      throw new NotFoundException();
+    }
+
+    if (question.creatorId !== user.id) {
+      throw new HttpException(
+        'You can only update your own questions',
+        HttpStatus.UNAUTHORIZED,
+      );
+    }
```

Comment:
should probably add a check so that students can't update their questions to forbidden statuses
```ts
// if you created the question (i.e. a student), you can't update the status to illegal ones
      if (
        body.status === asyncQuestionStatus.TADeleted ||
        body.status === asyncQuestionStatus.HumanAnswered
      ) {
        throw new HttpException(
          `You cannot update your own question's status to ${body.status}`,
          HttpStatus.UNAUTHORIZED,
        );
      }
```

Reply:
i did this